jekyll/_posts/guides/2018-09-21-usage-of-the-matrix-js-sdk.md

```diff
@@ -0,0 +1,199 @@
+
+Instantiate a new client object and use an `access token` to login:
+
+```javascript
+const client = sdk.createClient({
+    baseUrl: "https://matrix.org",
+    accessToken: "....MDAxM2lkZW50aWZpZXIga2V5CjAwMTBjaWQgZ2Vu....",
+    userId: "@USERID:matrix.org"
+});
+// note that we use the full MXID for the userId value
+```
+
+If you are logged into Riot, you can find an `access token` for the logged-in user on the Settings page.
+
+If the homeserver you're logging in to supports logging in with a password, you can also retrieve an `access token` programmatically using the API. To do this, create a new `client` with no authentication parameters, then call `client.login()` with `"m.login.password"`:
+
+```javascript
+const client = sdk.createClient("https://matrix.org");
+client.login("m.login.password", {"user": "USERID", "password":"hunter2"}).then((response) => {
+    console.log(response.access_token);
+});
+
+```
+
+In any case, once logged in either with a password or an access token, the client can get the current access token via:
+
+```javascript
+console.log(client.getAccessToken());
+```
+
+## Sync and Listen
+
+Let's perform a first sync, and listen for the response:
+
+```javascript
+client.once('sync', function(state, prevState, res) {
+    console.log(state); // state will be 'PREPARED' when the client is ready to use
+});
+```
+
+Once the sync is complete, we can add listeners for events. To listen to ALL incoming events:
+
+```javascript
+client.on("event", function(event){
+    console.log(event.getType());
+    console.log(event);
+})
+```
+
+But that would be a lot of traffic, and too much for our simple example. Let's just listen to events happening on the timeline of rooms for which our user is a member:
+
+```javascript
+client.on("Room.timeline", function(event, room, toStartOfTimeline) {
+    console.log(event.event);
+});
+```
+
+## Access the Store
+
+When we created a new client with `sdk.createClient()`, an instance of the default store, `MatrixInMemoryStore` was created and enabled.
+
+To access the store, we use accessor methods. For example, to get a list of rooms in which our user is joined:
+
+```javascript
+// client.client.getRooms() returns an array of room objects
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    console.log(room.roomId);
+});
+```
+
+More usefully, we could get a list of members for each of these rooms:
+
+```javascript
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    var members = room.getJoinedMembers();
+    members.forEach(member => {
+        console.log(member.name);
+    });
+});
+```
+
+For each room, we can inspect the timeline in the store:
+
+```javascript
+var rooms = client.getRooms();
+rooms.forEach(room => {
+    room.timeline.forEach(t => {
+        console.log(JSON.stringify(t.event.content));
+    });
+});
+```
+
+## Send messages to rooms
+
+To send a message, we create a content object, and specify a room to send to:
```

Comment:
might be fun for these (and others) to hyperlink to the right bit of the jsdoc or the spec so folks can see what the signature is?  e.g. i'm wondering what the 4th param to sendEvent is.

Reply:
Good idea